qiskit/primitives/gradient/lin_comb_estimator_gradient.py

```diff
@@ -0,0 +1,224 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Gradient of expectation values with linear combination of unitaries (LCU)
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Sequence, Type
+
+import numpy as np
+
+from qiskit import transpile
+from qiskit.circuit import (
+    Gate,
+    Instruction,
+    Parameter,
+    ParameterExpression,
+    QuantumCircuit,
+    QuantumRegister,
+)
+from qiskit.circuit.library.standard_gates import (
+    CXGate,
+    CYGate,
+    CZGate,
+    RXGate,
+    RXXGate,
+    RYGate,
+    RYYGate,
+    RZGate,
+    RZXGate,
+    RZZGate,
+)
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+from ..base_estimator import BaseEstimator
+from ..estimator_result import EstimatorResult
+from .base_estimator_gradient import BaseEstimatorGradient
+
+Pauli_Z = Pauli("Z")
+
+
+@dataclass
+class SubEstimator:
+    coeff: float | ParameterExpression
+    circuit: QuantumCircuit
+    index: int
+
+
+class LinCombEstimatorGradient(BaseEstimatorGradient):
+    """LCU estimator gradient"""
+
+    SUPPORTED_GATES = [
+        "rx",
+        "ry",
+        "rz",
+        "rzx",
+        "rzz",
+        "ryy",
+        "rxx",
+        "cx",
+        "cy",
+        "cz",
+        "ccx",
+        "swap",
+        "iswap",
+        "h",
+        "t",
+        "s",
+        "sdg",
+        "x",
+        "y",
+        "z",
+    ]
+
+    def __init__(
+        self, estimator: Type[BaseEstimator], circuit: QuantumCircuit, observable: SparsePauliOp
+    ):
+        self._circuit = circuit
+        self._observable = observable
+        self._grad, observable = self._preprocessing()
+        circuits = [self._circuit]
+        observables = [self._observable, observable]
+        for param, lst in self._grad.items():
+            for arg in lst:
+                circuits.append(arg.circuit)
+        super().__init__(estimator(circuits=circuits, observables=observables))
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        pass
+
+    @classmethod
+    def _gradient_circuits(cls, circuit: QuantumCircuit):
+        circuit2 = transpile(circuit, basis_gates=cls.SUPPORTED_GATES, optimization_level=0)
```

Comment:
Thank you for your information. I used `transpile` with level 0 optimization and `translator` [mode](https://github.com/Qiskit/qiskit-terra/blob/794db4792e97adca4a78eb4f3411c04e50665df7/qiskit/transpiler/preset_passmanagers/level0.py#L165). The `unroller` mode is not enough to translate `U` into `RY` and `RZ`, which is necessary for param shift, otherwise it results in wrong gradient.
Anyways, it would be nice if Terra can translate only parametrized gates. I would like use it too.